contracts/Strategy.sol

```diff
@@ -274,60 +362,312 @@ contract Strategy is BaseStrategy {
         (_liquidatedAmount, ) = liquidatePosition(type(uint256).max);
     }
 
-    // The _newStrategy must support the same kashiPair or bad things will happen
+    // The _newStrategy must support the same kashiPairs or bad things will happen
     function prepareMigration(address _newStrategy) internal override {
-        kashiPair.transfer(_newStrategy, kashiFraction());
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            liquidateKashiPair(
```

Comment:
we decided to avoid liquidations in migrations. if tokens can't be transferred, migration will not be used and strategy would be revoked + harvested to move to a second strat. Leave this method empty.

Reply:
The masterchef makes direct transfers difficult, because there's no callback on the new strategy to re-stake the tokens in the masterchef. If you have an idea how to deal with this, let me know.